File: test/fixtures/python/post_with_extra_whitespace.py
import requests

headers = {
    'accept': 'application/json',
    # requests won't add a boundary if this header is set when you pass files=
    # 'Content-Type': 'multipart/form-data',
}

files = {
    'files': open('47.htz', 'rb'),
    'name': (None, '47'),
    'oldMediaId': (None, '47'),
    'updateInLayouts': (None, '1'),
    'deleteOldRevisions': (None, '1'),
}

response = requests.post('http://localhost:28139/api/library', headers=headers, files=files)

File: test/fixtures/ruby/post_with_extra_whitespace.rb
require 'net/http'

uri = URI('http://localhost:28139/api/library')
req = Net::HTTP::Post.new(uri)
req['accept'] = 'application/json'

req.set_form(
  [
    [
      'files',
      File.open('47.htz')
    ],
    [
      'name',
      '47'
    ],
    [
      'oldMediaId',
      '47'
    ],
    [
      'updateInLayouts',
      '1'
    ],
    [
      'deleteOldRevisions',
      '1'
    ]
  ],
  'multipart/form-data'
)

req_options = {
  use_ssl: uri.scheme == 'https'
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end
